### PR TITLE
feat(NumberInput): add Int validation for Form.NumberInput

### PR DESCRIPTION
**Pull request by inemiro**
[FX-1994]

### Description

Validation is required, because user can input custom value even if max and min are specified.
Form.NumberInput should check if value is in `Int` range between  [−2,147,483,648 to 2,147,483,647].
Readable user error message have to be shown.

### How to test

Try to input or paste some huge number value
https://picasso.toptal.net/?path=/story/picasso-forms-form--form
in comparison to
https://picasso.toptal.net/FX-1994-add-int-validation-form-input/?path=/story/picasso-forms-form--form

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

[FX-1994]: https://toptal-core.atlassian.net/browse/FX-1994

### CODE REVIEW

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
-    {(inputProps: NumberInputProps) => {
-      return <PicassoNumberInput {...inputProps} />
-    }}
-  </FieldWrapper>
-)
+const MAX_INT = 2147483648
+
+export const NumberInput = (props: Props) => {
+  const { min = -MAX_INT, max = MAX_INT, validate } = props
+  const wrappedValidate: FieldValidator<NumberInputProps['value']> = useCallback(
```
Comment (OleksandrNechai) [inline packages/picasso-forms/src/NumberInput/NumberInput.tsx:16]: Why do we need `useCallback`?
Can we maybe compose validators with https://github.com/toptal/picasso/blob/master/packages/picasso-forms/src/utils/validators.ts#L1

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
 export type Props = NumberInputProps & FieldProps<NumberInputProps['value']>
 
-export const NumberInput = (props: Props) => (
-  <FieldWrapper<NumberInputProps> {...props}>
-    {(inputProps: NumberInputProps) => {
-      return <PicassoNumberInput {...inputProps} />
-    }}
-  </FieldWrapper>
-)
+const MAX_INT = 2147483648
```
Comment (OleksandrNechai) [inline packages/picasso-forms/src/NumberInput/NumberInput.tsx:12]: Are you sure that acceptable negative and positive numbers are the same?

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
+        return `Must be greater than or equal to ${min}.`
+      }
+
+      return validate?.(...args)
+    },
+    [min, max, validate]
+  )
+
+  return (
+    <FieldWrapper<NumberInputProps> {...props} validate={wrappedValidate}>
```
Comment (OleksandrNechai) [inline packages/picasso-forms/src/NumberInput/NumberInput.tsx:33]: `wrappedValidate` sounds weird to me :smile: Maybe `validate={composeValidators([validateNumberLimits,validate])}`?

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx [hunk 4]
```diff
@@ -1,20 +1,49 @@
...
+      }
+    },
+    [min, max]
+  )
+
+  return (
+    <FieldWrapper<NumberInputProps>
+      {...props}
+      validate={composeValidators(
+        [validateNumberLimits, validate].filter(Boolean)
```
Comment (OleksandrNechai) [inline packages/picasso-forms/src/NumberInput/NumberInput.tsx:38]: How about putting this line inside `composeValidators`? :)

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx [hunk 5]
```diff
@@ -1,20 +1,49 @@
...
-  </FieldWrapper>
-)
+const MIN_INT = 2147483648
+const MAX_INT = 2147483647
+
+const { composeValidators } = validators
+
+export const NumberInput = (props: Props) => {
+  const { min = -MIN_INT, max = MAX_INT, validate } = props
+  const validateNumberLimits: FieldValidator<NumberInputProps['value']> = useCallback(
```
Comment (OleksandrNechai) [inline packages/picasso-forms/src/NumberInput/NumberInput.tsx:20]: Oh, come on, will performance degrade if we don't use `useCallback`? :thinking:

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx [hunk 6]
```diff
@@ -1,20 +1,49 @@
...
 
-export const NumberInput = (props: Props) => (
-  <FieldWrapper<NumberInputProps> {...props}>
-    {(inputProps: NumberInputProps) => {
-      return <PicassoNumberInput {...inputProps} />
-    }}
-  </FieldWrapper>
-)
+const MIN_INT = 2147483648
+const MAX_INT = 2147483647
```
Comment (OleksandrNechai) [inline packages/picasso-forms/src/NumberInput/NumberInput.tsx:14]:
```suggestion
const MIN = -2147483648
const MAX = 2147483647
```

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx [hunk 7]
```diff
@@ -1,20 +1,49 @@
...
+
+const { composeValidators } = validators
+
+export const NumberInput = (props: Props) => {
+  const { min = -MIN_INT, max = MAX_INT, validate } = props
+  const validateNumberLimits: FieldValidator<NumberInputProps['value']> = useCallback(
+    (...args) => {
```
Comment (OleksandrNechai) [inline packages/picasso-forms/src/NumberInput/NumberInput.tsx:21]:
```suggestion
    ([value]) => {
```

File: packages/picasso-forms/src/NumberInput/NumberInput.tsx (hunk 4 above)
Comment (OleksandrNechai) [reply packages/picasso-forms/src/NumberInput/NumberInput.tsx:38]: I mean `.filter(Boolean)`

**Review by OleksandrNechai — APPROVED** [review]
:+1:  thanks! :rose:

**Review by PeterMK85 — APPROVED** [review]

**Review by viktorbezdek — COMMENTED** [review]
👍🏻

**Pull request merged**